_notebooks/2020-07-28-lineage.ipynb

```diff
@@ -0,0 +1,323 @@
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "METADATA_STORE_HOST = \"metadata-grpc-service.kubeflow\" # default DNS of Kubeflow Metadata gRPC serivce.\n",
+    "METADATA_STORE_PORT = 8080"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a workspace and a run in a workspace\n",
+    "* [Workspace](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L92) groups a set of runs of pipelines, notebooks and their related artifacts and executions\n",
+    "* [Store](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L59) that connects to the Metadata gRPC service\n",
+    "* [Run](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L227) captures a run of pipeline or notebooks in a workspace and group execution"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ws1 = metadata.Workspace(\n",
+    "    # Connect to metadata service in namespace kubeflow in k8s cluster.\n",
+    "    store=metadata.Store(grpc_host=METADATA_STORE_HOST, grpc_port=METADATA_STORE_PORT),\n",
+    "    name=\"xgboost-synthetic\",\n",
+    "    description=\"workspace for xgboost-synthetic artifacts and executions\",\n",
+    "    labels={\"n1\": \"v1\"})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "r = metadata.Run(\n",
+    "    workspace=ws1,\n",
+    "    name=\"xgboost-synthetic-faring-run\" + datetime.utcnow().isoformat(\"T\") ,\n",
+    "    description=\"a notebook run\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create an execution in a run\n",
+    "* [Execution](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L251) captures a run of pipeline or notebooks in a workspace and group executions. Execution also serves as object for logging artifacts as its input or output"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": "An execution was created with id 290\n"
+    }
+   ],
+   "source": [
+    "exec = metadata.Execution(\n",
+    "    name = \"execution\" + datetime.utcnow().isoformat(\"T\") ,\n",
+    "    workspace=ws1,\n",
+    "    run=r,\n",
+    "    description=\"execution for training xgboost-synthetic\",\n",
+    ")\n",
+    "print(\"An execution was created with id %s\" % exec.id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Log a data set and a model\n",
```

Comment:
`DataSet` and `Model` are subclasses of `Artifact`. So the SDK provides some ready-to-use APIs for common Artifacts. These classes have specific fields (like `query` or `hyperparameters`). It would be nice to have a paragraph that describes what are the available Artifacts and if/how one could create new ones.